### PR TITLE
pnpm@8 8.15.7 (new formula)

### LINKED ISSUE #6903: Support shipping pnpm via brew with versioned installs
For other formulas on brew, you have the ability to install specific versions. For example, if you want to install an older version of `node` you can see there are options to install

```sh
❯ brew search node
==> Formulae
libbitcoin-node        node                   node@14                node_exporter          nodenv
linode-cli             node-build             node@16                nodebrew               ode
llnode                 node-sass              node@18                nodeenv
```

For pnpm however there is not this ability, which makes downgrading or pinning engineers to a specific version difficult. While my team plans to use `corepack` to set the pnpm version, there are other hurdles that prevent us from doing that in the near term. Having the ability to `brew install pnpm@VERSION` would be helpful.

You can see example formulas here for Node

https://formulae.brew.sh/api/formula/node@18.json
https://formulae.brew.sh/formula/node

Having something similar for pnpm would be helpful

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

As a developer, I'd like to give my developers the ability to install a specific versions of `pnpm` via `brew install`

**Describe the solution you'd like**

I'd like to give my developers the ability to do `brew install pnpm@8` or `brew install pnpm@7`

**Describe the drawbacks of your solution**

If there is a breaking lockfile change (like we had from 8.6.0 to 8.6.10) we can't force people to fix forward. This is where we intend to leverage `corepack` going forward.

**Describe alternatives you've considered**

`corepack` to encode `packageManager` into the package.json. Then have tools installed to force this to run when you enter a directory and auto set the node and pnpm version for the user

### PR DESCRIPTION
**Pull request by Pandoks**
**Checklist:**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

**Description:**
- Adds versioning to `pnpm` similar to `node` (ie. `node@20`, `node@18`, etc)
- Adds _only_ `pnpm@8` as a `keg_only` formula
- Resolves pnpm/pnpm#6903

### CODE REVIEW

**Comment by SMillerDev** [comment]
Please indicate how this would fill our criteria for versioned formulas (https://docs.brew.sh/Versions) and why we might want to ship an older version?

**Comment by SMillerDev** [comment]
Thanks for the explanation, is there any indication until when it is supported?

**Comment by Pandoks** [comment]
> Thanks for the explanation, is there any indication until when it is supported?

when what is supported?

**Comment by SMillerDev** [comment]
> when what is supported?

Until when this version of pnpm is supported

**Comment by Pandoks** [comment]
> > when what is supported?
> 
> Until when this version of pnpm is supported

This version of pnpm is supported (version 8). I mentioned that version 7 isn't supported because I said `Will submit another pull request for pnpm@7 if pnpm@8 gets accepted` in my pull request. (Removed now)

**Comment by SMillerDev** [comment]
> This version of pnpm is supported (version 8). I mentioned that version 7 isn't supported because I said Will submit another pull request for pnpm@7 if pnpm@8 gets accepted in my pull request. (Removed now)

But in what term is version 8 supported? Until the inevitable heat death of the universe? Until next week?

**Comment by Pandoks** [comment]
> > This version of pnpm is supported (version 8). I mentioned that version 7 isn't supported because I said Will submit another pull request for pnpm@7 if pnpm@8 gets accepted in my pull request. (Removed now)
> 
> But in what term is version 8 supported? Until the inevitable heat death of the universe? Until next week?

1 year

**Comment by SMillerDev** [comment]
Could you add a `disable!` with a date when it stops being supported then

File: Formula/p/pnpm@8.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    sha256 cellar: :any,                 arm64_sonoma:   "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 arm64_ventura:  "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 arm64_monterey: "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 sonoma:         "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any,                 ventura:        "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any,                 monterey:       "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "ea36ce48c3435f17c6c3f59ca1f0113d7111f5ddb15ed665d2b2265fb374e816"
+  end
+
+  disable! date: "2025-03-20", because: :unmaintained
```
Comment (chenrui333) [inline Formula/p/pnpm@8.rb:25]: any upstream release policy on this?

File: Formula/p/pnpm@8.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  bottle do
+    sha256 cellar: :any,                 arm64_sonoma:   "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 arm64_ventura:  "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 arm64_monterey: "f8d7d13faf42b5735020046a055c7222e62fc1ec8baa7e342a7238ff8317af59"
+    sha256 cellar: :any,                 sonoma:         "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any,                 ventura:        "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any,                 monterey:       "c72c7e5b336c8009c5965d1fdd4af07a46bafab5b0e3126c8f48781caebe6002"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "ea36ce48c3435f17c6c3f59ca1f0113d7111f5ddb15ed665d2b2265fb374e816"
+  end
```
Comment (chenrui333) [inline Formula/p/pnpm@8.rb:23]: for new formula, no need to have this
```suggestion
```

**Review by chenrui333 — APPROVED** [review]

File: Formula/p/pnpm@8.rb (hunk 1 above)
Comment (Pandoks) [reply Formula/p/pnpm@8.rb:17]: I emailed them

Comment (SMillerDev) [reply Formula/p/pnpm@8.rb:17]: Could you ask that they also document that somewhere publicly? I doubt we're the only people to wonder about this.

**Comment by Pandoks** [comment]
Got them to include the date on their policy page:
[https://github.com/pnpm/pnpm/security/policy](https://github.com/pnpm/pnpm/security/policy)

File: Formula/p/pnpm@8.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/p/pnpm@8.rb]: yeah, it would be better document publicly and good for the community.

Comment (chenrui333) [reply Formula/p/pnpm@8.rb]: oh, i see your security policy comment, all good.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@Pandoks, thanks for your first contribution to homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**